vumi/tests/helpers.py

```diff
@@ -1268,6 +1276,13 @@ def cleanup(self):
 
         self._unpatch()
 
+        if unclosed_managers and self._assert_closed:
+            # We have unclosed managers and we've been asked to assert that we
+            # don't.
+            raise Exception(
```

Comment:
Goog point. That'll make the test for this behaviour more specific, if nothing else.